src/main/java/org/snomed/snowstorm/mrcm/MRCMService.java

```diff
@@ -112,13 +111,18 @@
 				}
 			}
 		}
-		for (ConceptMini attributeConceptMini : attributeConceptMinis) {
+		addExtraConceptMiniFields(attributeDomains, branchMRCM, attributeConceptMinis);
+
+		return attributeConceptMinis;
+	}
+
+	private void addExtraConceptMiniFields(final List<AttributeDomain> attributeDomains, final MRCM branchMRCM, final Collection<ConceptMini> attributeConceptMinis) {
+		attributeConceptMinis.forEach(attributeConceptMini -> {
 			attributeConceptMini.addExtraField("attributeDomain",
 					attributeDomains.stream().filter(attributeDomain -> attributeConceptMini.getId().equals(attributeDomain.getReferencedComponentId()))
 							.collect(Collectors.toSet()));
-		}
-
-		return attributeConceptMinis;
+			attributeConceptMini.addExtraField("attributeRange", branchMRCM.getAttributeRanges());
```

Comment:
It looks like this adds all attribute ranges to every attributeDomain.
Please only add the attribute ranges applicable to each attributeDomain.
I think the attributeRange.referencedComponentId should match the attributeDomain.referencedComponentId so you could use a filter here.

Reply:
Fixed following the comments we talked about in chat:

`we should keep the list of AttributeRanges separate in the response and just filter the list based on the AttributeDomains and content type that has been selected.`